File: api/src/main/java/fr/huiitre/tools/modules/core/admin/application/usecase/GetAdminStatsUseCase.java
package fr.huiitre.tools.modules.core.admin.application.usecase;

import fr.huiitre.tools.modules.core.admin.application.ports.AdminStatsRepository;
import fr.huiitre.tools.modules.core.admin.application.view.AdminStatsView;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;
import org.springframework.stereotype.Service;

@Service
public class GetAdminStatsUseCase implements SecuredUseCase {

    @Override
    public RoleCode requiredRole() {
        return RoleCode.ADMIN;
    }

    private final AdminStatsRepository adminStatsRepository;

    public GetAdminStatsUseCase(AdminStatsRepository adminStatsRepository) {
        this.adminStatsRepository = adminStatsRepository;
    }

    public AdminStatsView execute() {
        return adminStatsRepository.getStats();
    }
}

File: api/Modules/Core/Admin/Application/Usecases/GetAdminStatsUseCase.cs
using Tools.Api.Modules.Core.Admin.Application.Dto;
using Tools.Api.Modules.Core.Admin.Application.Ports;
using Tools.Api.Modules.Core.Security.Application.Ports;
using Tools.Api.Modules.Core.Security.Application.Services;
using Tools.Api.Modules.Core.Security.Application.Usecases;
using Tools.Api.Modules.Core.Security.Domain;

namespace Tools.Api.Modules.Core.Admin.Application.Usecases;

// Cas d'usage administrateur : indicateurs du tableau de bord.
public sealed class GetAdminStatsUseCase(
    UseCaseAuthorizer authorizer,
    IAdminStatsRepository adminStatsRepository
) : SecuredUseCase(authorizer)
{
    protected override RoleCode RequiredRole => RoleCode.Admin;

    public Task<AdminStatsDto> Execute()
    {
        return adminStatsRepository.GetStatsAsync();
    }
}
